### ISSUE TITLE
Plan release path towards stable

### ISSUE THREAD

**Issue by felixarntz** [WordPress/performance#253]
Today, during the performance Slack chat we discussed the topic of how many more beta releases, and then RC releases, we want to have before the eventual `1.0.0` stable. This issue's purpose is to finalize the plan. [See relevant Slack chat log for more context](https://wordpress.slack.com/archives/C02KGN5K076/p1647965634404359)

One thing that was decided was that, at least until the stable release, we'll publish a new version every 2 weeks, similar to how we've done so far, with the next release being on **April 4**.

Please keep in mind that the "beta" and "rc" suffixes relate primarily to the overall plugin infrastructure. The individual modules have their own stages of development, which isn't necessarily captured in the suffix right now. Some modules (e.g. the experimental ones) are probably more "beta" than others, just as an example. But we should definitely bump the overall plugin to a stable version regardless of the state the modules are in - especially since this is a beta testing plugin anyway.

The Slack chat discussion converged towards the following proposal:
* `1.0.0-rc.1` on **April 4**
* `1.0.0` stable on **April 18**

In other words, the next release would be the Release Candidate, and since we already had an extensive beta, we can bump to stable after that.

**Comment by felixarntz** [comment]
_Please vote on the following by **Monday, March 28, 2022**:_

> The Slack chat discussion converged towards the following proposal:
> * `1.0.0-rc.1` on **April 4**
> * `1.0.0` stable on **April 18**

**If you agree, please give a 👍 reaction. Otherwise a 👎.**
If the latter, it would be great if you could also leave a comment with some more context or an alternative proposal.

**Comment by mxbclang** [comment]
**Voting is now closed.**

Based on **6** thumbs up votes and **zero** thumbs down votes, we'll proceed with this behavior:

- Release `1.0.0-rc.1` on April 4
- Release `1.0.0` stable on April 18

**Comment by felixarntz** [comment]
New `release/1.0.0-rc.1` branch has been created, with #253 to merge the additional enhancements from the (now temporary) `release/1.0.0-beta.4` branch into it.

Will also adjust the relevant PRs to go against the new RC branch.
